core/vm/intpool.go

```diff
@@ -65,3 +69,46 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolDefaultCap = 5
+
+// IntPoolPool manages a pool of intPools
+type intPoolPool struct {
+	pools []*intPool
+
+	// Mutex
+	mutex sync.Mutex
+}
+
+func initPool() {
+	poolOfIntPools = intPoolPool{
+		pools: make([]*intPool, 0, poolDefaultCap),
+	}
+}
+
+var poolOfIntPools intPoolPool
+
+// get is looking for an available pool to return
+func (ipp *intPoolPool) get() *intPool {
+	ipp.mutex.Lock()
+	defer ipp.mutex.Unlock()
+
+	for _, ip := range ipp.pools {
+		if ip.inUse == false {
+			ip.inUse = true
+			return ip
+		}
+	}
```

Comment:
I don't get why you're doing this boolean flag thing. Just keep the unused pools in here. When someone requests it, give it to them and remove it from the slice. When they return it, add it back to the slice (as long as it doesn't overflow).